source/frontend/src/utils/axiosUtils.ts

```diff
@@ -47,10 +49,13 @@ export function useAxiosErrorHandlerWithAuthorization(
     (axiosError: AxiosError<IApiError>) => {
       if (axiosError?.response?.status === 400) {
         toast.error(axiosError?.response.data.error, { autoClose: 10000 });
```

Comment:
this is an example, when the error is handled satisfactorily, we resolve the promise. when we get an unexpected error, we reject and fall back on the default error logging behaviour.